fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreCountRecordsTest.java

```diff
@@ -71,6 +81,9 @@
 @Tag(Tags.RequiresFDB)
 public class FDBRecordStoreCountRecordsTest extends FDBRecordStoreTestBase {
```

Comment:
I may be missing it, but this could probably use a test of `deleteRecordsWhere` that fails if the record count key isn't aligned, checking with various states of the record count key.

Reply:
Yes, I think you are correct, that is not covered, and it would be good to add that test.